wp-content/civi-extensions/goonjcustom/Civi/CollectionCampService.php

```diff
@@ -492,30 +492,36 @@ public static function generateCollectionCampQr(string $op, string $objectName,
     }
 
     $collectionCamps = EckEntity::get('Collection_Camp', TRUE)
-      ->addSelect('Collection_Camp_Core_Details.Status', 'Collection_Camp_Core_Details.Contact_Id')
+      ->addSelect('Collection_Camp_Core_Details.Status', 'Collection_Camp_Core_Details.Contact_Id', 'subtype:name')
```

Comment:
I've added a check to ensure `$collectionCampSubtype` is not null or empty before using it. If it's missing, a Civi log warning is recorded, and the process stops early to prevent generating incorrect QR codes.